kubernetes/config/kube_config_test.py

```diff
@@ -0,0 +1,413 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import os
+import tempfile
+import unittest
+
+from .incluster_config import ConfigException
+from .kube_config import ConfigNode, FileOrData, KubeConfigLoader
+
+
+def _base64(string):
+    return base64.encodestring(string.encode()).decode()
+
+
+TEST_FILE_KEY = "file"
+TEST_DATA_KEY = "data"
+TEST_FILENAME = "testfilename"
+
+TEST_DATA = "testdata"
+TEST_DATA_BASE64 = _base64(TEST_DATA)
+
+TEST_ANOTHER_DATA = "anothertestdata"
+TEST_ANOTHER_DATA_BASE64 = _base64(TEST_ANOTHER_DATA)
+
+TEST_HOST = "testhost"
+TEST_USERNAME = "me"
+TEST_PASSWORD = "pass"
+# token for me:pass
+TEST_BASIC_TOKEN = "Basic bWU6cGFzcw=="
+
+TEST_SSL_HOST = "https://testhost"
+TEST_CERTIFICATE_AUTH = "certauth"
+TEST_CERTIFICATE_AUTH_BASE64 = _base64(TEST_CERTIFICATE_AUTH)
+TEST_CLIENT_KEY = "clientkey"
+TEST_CLIENT_KEY_BASE64 = _base64(TEST_CLIENT_KEY)
+TEST_CLIENT_CERT = "clientcert"
+TEST_CLIENT_CERT_BASE64 = _base64(TEST_CLIENT_CERT)
+
+
+class TestRoot(unittest.TestCase):
+
+    def setUp(self):
+        self._temp_files = []
+
+    def tearDown(self):
+        for f in self._temp_files:
+            os.remove(f)
+
+    def _create_temp_file(self, content=""):
+        handler, name = tempfile.mkstemp()
+        self._temp_files.append(name)
+        os.write(handler, str.encode(content))
+        os.close(handler)
+        return name
+
+
+class TestFileOrData(TestRoot):
+
+    def get_file_content(self, file):
```

Comment:
Looked at this briefly, we have pep8 that is run as part of scripts/update-pep8.sh but not in tox. It also does not catch these kind of problems. I've created an issue for this: #52